### PR TITLE
[test/plugin] Enhance PluginTest

### PR DESCRIPTION
**Pull request by dmsolr**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance
- [x] Enhancement

- Related issues

___
### New feature or improvement
1. Re-balance the workload of plugintests
2. Restore testcases of MySQL, remove on  #4641 
3. Support user to runs plugin-test on debug mode. It will not remove the log files and actualData.yaml 
4. remove paramater 'parallel_run_size' and disable parallel test.

### CODE REVIEW

**Comment by kezhenxu94** [comment]
> Re-balance the workload of plugintests

I don't think we need this and you actually cannot "rebalance" them, as we're using matrix build, all the plugin tests of each plugin are run parallelly , no mater which workflow they belong to

File: .github/workflows/plugins-test.0.yaml [hunk 1]
```diff
@@ -48,9 +48,9 @@ jobs:
           - { name: 'elasticsearch-6.x-scenario', title: 'ElasticSearch 6.7.1-6.8.4 (7)' }
           - { name: 'elasticsearch-7.x-scenario', title: 'ElasticSearch 7.0.0-7.5.2 (14)' }
           - { name: 'feign-scenario', title: 'Feign 9.0.0-9.5.1 (8)' }
-          - { name: 'finagle-17.10.x-scenario', title: 'Finagle 17.10.0-20.1.0' }
-          - { name: 'finagle-6.25.x-scenario', title: 'Finagle 6.25.0-6.43.0' }
-          - { name: 'finagle-6.44.x-scenario', title: 'Finagle 6.44.0-7.1.0' }
+          - { name: 'finagle-17.10.x-scenario', title: 'Finagle 17.10.0-20.1.0(19)' }
+          - { name: 'finagle-6.25.x-scenario', title: 'Finagle 6.25.0-6.43.0 (18)' }
+          - { name: 'finagle-6.44.x-scenario', title: 'Finagle 6.44.0-7.1.0 (4)' }
```
Comment (kezhenxu94) [inline .github/workflows/plugins-test.0.yaml:53]: I personally don't like these numbers `(19), (18), (4)`, somewhat pointless yet increase maintenance work, WDYT? @wu-sheng

**Comment by kezhenxu94** [comment]
> > Re-balance the workload of plugintests
> 
> I don't think we need this and you actually cannot "rebalance" them, as we're using matrix build, all the plugin tests of each plugin are run parallelly , no mater which workflow they belong to

The time it costs only depends on the version numbers of the plugin test and the test itself

File: .github/workflows/plugins-test.0.yaml (hunk 1 above)
Comment (wu-sheng) [reply .github/workflows/plugins-test.0.yaml:53]: I am OK in either way. No prefer. Maintaining this is good for rebalancing only.

Comment (kezhenxu94) [reply .github/workflows/plugins-test.0.yaml:53]: > I am OK in either way. No prefer. Maintaining this is good for rebalancing only.

We don't and can't rebalance them now, all plugin tests run in parallel, saying "rebalance" makes no sense at all

Comment (wu-sheng) [reply .github/workflows/plugins-test.0.yaml:53]: I don't feel much objective about this word `rebalance`. From my understanding, rebalancing could target time or number of group elements. For time, yes, they are parallel, but if one fails, the number of group elements matters when we re-run.

Basically you could consider my point as `no harm at least`.

Comment (kezhenxu94) [reply .github/workflows/plugins-test.0.yaml:53]: For rerunning failed tests, make sense to me now, didn’t thought about this

Comment (wu-sheng) [reply .github/workflows/plugins-test.0.yaml:53]: OK. Since the concerns have be resolved, merging.

Comment (dmsolr) [reply .github/workflows/plugins-test.0.yaml:53]: 1. I agree with it is hard to maintain the number of versions. And now, I still cannot sure they are correct in all.
2. Make every group have the same number of versions as far as possible. It is necessary. If not, we don't need to split them into multi groups. At least, the adjustment is harmless.

**Pull request merged**